### ISSUE TITLE
Destructuring assignments are producing errors with noImplicitAny

### ISSUE THREAD

**Issue by thebanjomatic** [Microsoft/TypeScript#8229]
I found an issue in the current nightly builds relating to destructuring assignments with noImplicitAny enabled. The actual code was more complicated, but it can be simplified to the following:

```typescript
let [a, b, c] = [1, 2, 3];
```

The variables appear to be typed correctly on hover, however, 3 errors are produced inside the assignment stating that `a`, `b`, and `c` have an implicit type of any. 

It can be silenced by the following, but that's not ideal:
```typescript
let [a, b, c] = <[number, number, number]>[1, 2, 3];
```

**TypeScript Version:** 

Searching backwards, it first appears in:  
* nightly (1.9.0-dev.20160409)

**Expected behavior:**

There shouldn't be an implicit any error in this case.

**Actual behavior:** 

An erroneous error is produced.

**Comment by mhegazy** [comment]
looks like a regression from https://github.com/Microsoft/TypeScript/issues/6935

**Comment by thebanjomatic** [comment]
@sheetalkamat Any update on this issue? I came across the problem again when trying to build my code-base with the 2.0 beta.

**Comment by AlexGalays** [comment]
This bug is going to be encountered fairly frequently using TS 2.0 beta
